pkg/codegen/internal/test/testdata/simple-methods-schema/dotnet/Foo.cs

```diff
@@ -0,0 +1,150 @@
+// *** WARNING: this file was generated by test. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Pulumi.Serialization;
+
+namespace Pulumi.Example
+{
+    [ExampleResourceType("example::Foo")]
+    public partial class Foo : Pulumi.ComponentResource
+    {
+        /// <summary>
+        /// Create a Foo resource with the given unique name, arguments, and options.
+        /// </summary>
+        ///
+        /// <param name="name">The unique name of the resource</param>
+        /// <param name="args">The arguments used to populate this resource's properties</param>
+        /// <param name="options">A bag of options that control this resource's behavior</param>
+        public Foo(string name, FooArgs? args = null, ComponentResourceOptions? options = null)
+            : base("example::Foo", name, args ?? new FooArgs(), MakeResourceOptions(options, ""), remote: true)
+        {
+        }
+
+        private static ComponentResourceOptions MakeResourceOptions(ComponentResourceOptions? options, Input<string>? id)
+        {
+            var defaultOptions = new ComponentResourceOptions
+            {
+                Version = Utilities.Version,
+            };
+            var merged = ComponentResourceOptions.Merge(defaultOptions, options);
+            // Override the ID if one was specified for consistency with other language SDKs.
+            merged.Id = id ?? merged.Id;
+            return merged;
+        }
+
+        /// <summary>
+        /// A description of bar.
+        /// </summary>
+        public Pulumi.Output<FooBarResult> Bar(FooBarArgs args)
```

Comment:
Should this be `FooBarArgs? args` here? Since `args ?? new FooBarArgs()` is trying to deal with nullables?

Reply:
No, the signature is right. There are required inputs for this method, so it should be `FooBarArgs`. The `args ?? new FooBarArgs()` is the same thing we do for functions:

https://github.com/pulumi/pulumi/blob/4a035c44aa80a31be0af942db438b430e3ed3a57/pkg/codegen/dotnet/gen.go#L1116

While the C# compiler will help prevent you from passing `null` here with warnings, someone could force it, or a null value could come in from another language, so this is ensures it's initialized to an instance of the appropriate `Args` class. When the args class is serialized, it will throw errors if a required input isn't set:

https://github.com/pulumi/pulumi/blob/4a035c44aa80a31be0af942db438b430e3ed3a57/sdk/dotnet/Pulumi/Resources/InputArgs.cs#L57-L60